src/Proxy/DriveItemProxy.php

```diff
@@ -255,45 +257,86 @@ public function delete()
      * Uploads a file under this folder drive item.
      *
      * @param string $name
-     *        The name.
-     * @param string|resource|\GuzzleHttp\Psr7\Stream $content
-     *        The content.
+     *        The name of file(relative file path if in same directory (or) absolute file path).
      * @param array $options
      *        The options.
      *
      * @return DriveItemProxy
      *         The drive item created.
      *
-     * @todo Support name conflict behavior.
-     * @todo Support content type in options.
      */
-    public function upload($name, $content, array $options = [])
+    public function upload($name, array $options = [])
     {
         $name         = rawurlencode($name);
         $driveLocator = "/drives/{$this->parentReference->driveId}";
         $itemLocator  = "/items/{$this->id}";
-        $endpoint     = "$driveLocator$itemLocator:/$name:/content";
-
-        $body = $content instanceof Stream ?
-            $content
-            : Psr7\stream_for($content);
-
-        $response = $this
+        $uploadSessionEndpoint = "$driveLocator$itemLocator:/$name:/createUploadSession";
+        //include extended options with existing item for upload session
+        $item = array_merge(
+            array("name" => $name, 'description' => '' ,"@microsoft.graph.conflictBehavior" => "rename"), $options
```

Comment:
Use `[]` for arrays, and split multiple values on multiple lines.

Also, please could you make the conflict behavior configurable via `$options`?

Reply:
Yep. Can I have it like this,
 `$item = $options + [
    'name' => $name,
    'description' => '' ,
    '@microsoft.graph.conflictBehavior' => 'rename'  //default conflict behaviour
];`

The later part is added only when they are not mentioned. Omitted if already there as mentioned like in here, http://php.net/manual/en/language.operators.array.php

Reply:
done